Sage/carbon
# refactor: replace element-resize with useResizeObserver - FE-4290

### Proposed behaviour
Remove ElementResize util in favor of useResizeObserver hook

### Checklist



### Testing instructions
Check following stories locally: 
Sticky footer functionality in Form component  - http://localhost:9001/?path=/docs/design-system-form--default-with-sticky-footer
Dialog with dynamic content (center dialog after resize) - http://localhost:9001/?path=/story/dialog--dynamic-content
